### ISSUE TITLE
External editor

### ISSUE THREAD

**Issue by kupiqu** [flathub/org.octave.Octave#105]
I cannot set Kate as external editor in Octave. Is there a way to make /usr/bin/kate accessible?

It doesn't work either when calling kate using `system`, which I assume is basically the same issue.

**Comment by mtmiller** [comment]
TL;DR - no, you can't make anything in /usr/bin accessible to any Flatpak app.

You can expose certain parts of the host filesystem to the Flatpak app using the --filesystem option, so if you have a copy of Kate installed in /opt or in /usr/local, that could be made to work. But since Kate is a KDE based app, it's possible that the KDE libs in the Flatpak runtime won't be compatible with the KDE / Qt version that your host copy of Kate uses.

I _believe_ if you use `system ('xdg-open file.txt')`, the xdg-open program is specifically designed to work across the Flatpak-to-host bridge and call the appropriate helper program on your host, or even another Flatpak app.

**Comment by kupiqu** [comment]
> I believe if you use system ('xdg-open file.txt'), the xdg-open program is specifically designed to work across the Flatpak-to-host bridge and call the appropriate helper program on your host, or even another Flatpak app.

Interesting. It sort of works but not quite. using xdg-open outside flatpak opens kate properly, but inside octave (via system) requests which app to use.

I think it has to do with mimeapps.list that is not accessible to flatpak. Any suggestion of how to deal with this?

**Comment by mtmiller** [comment]
I get the same dialog when using `xdg-open` in any flatpak context. I'm not sure if there is a way to configure that, but that's a general flatpak issue.

**Comment by kupiqu** [comment]
It may be a bug, as it can be configured outside flatpak. Will open an issue in the flatpak repo... thank you.

**Comment by kupiqu** [comment]
As mentioned in https://github.com/flatpak/flatpak/issues/3601:

> After you select the same application a few times it will stop asking

**Comment by mtmiller** [comment]
Thanks, that's useful to know
